### PR TITLE
Set Property(required) rule not working

### PR DESCRIPTION
**Pull request by gargadobe**
Partial Fix for  [#211](https://github.com/adobe-rnd/form-block/issues/211)

Test URLs:
- Before: https://main--aem-boilerplate-forms--adobe-rnd.hlx.live/drafts/garg/form
- After: https://issue221--aem-boilerplate-forms--adobe-rnd.hlx.live/drafts/garg/form

Fix:

In given form , on change of first text input, second number input should become mandatory as per rules which was not working earlier

URL for testing:

-URL for testing:
-  https://issue211--aem-boilerplate-forms--adobe-rnd.hlx.page/drafts/garg/form

### CODE REVIEW

File: blocks/form/rules/index.js [hunk 1]
```diff
@@ -34,11 +34,8 @@ async function fieldChanged(payload, form, generateFormRendition) {
     }
     switch (propertyName) {
       case 'required':
-        if (currentValue === true) {
-          field.closest('.field-wrapper').dataset.required = '';
-        } else {
-          field.closest('.field-wrapper').removeAttribute('data-required');
-        }
+        field.toggleAttribute('required', currentValue === true);
+        field.closest('.field-wrapper').setAttribute('data-required', currentValue === true);
```
Comment (vdua) [inline blocks/form/rules/index.js:37]: test case.

File: test/fixtures/dynamic/update-required.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+export const sample = {
+  items: [
+    {
+      fieldType: 'text-input',
+      id: 'text-input',
+      name: 'name',
+      events: {
+        change: [
+          "if(contains($event.payload.changes[].propertyName, 'value'), [dispatchEvent(numberinput, 'custom:setProperty', {required : true()}), dispatchEvent(checkboxgroup, 'custom:setProperty', {required : true()}) ,  dispatchEvent(dropdown, 'custom:setProperty', {required : true()}),  dispatchEvent(radiogroup, 'custom:setProperty', {required : true()})],{})",
+        ],
```
Comment (vdua) [inline test/fixtures/dynamic/update-required.js:14]: I would recommend enhancing that test case to test the opposite.

Comment (gargadobe) [reply test/fixtures/dynamic/update-required.js:14]: what do you mean by opposite ? default true and then make false ?

Comment (vdua) [reply test/fixtures/dynamic/update-required.js:14]: yes

**Comment by ravise5** [comment]
This is already fixed.

**Pull request closed without merging**